crates/tako/src/server/monitoring.rs

```diff
@@ -0,0 +1,66 @@
+use crate::messages::gateway::{CollectedOverview, LostWorkerReason};
+use crate::messages::worker::WorkerOverview;
+use crate::WorkerId;
+
+use crate::messages::common::WorkerConfiguration;
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+
+pub struct EventHistory {
+    event_queue: Vec<MonitoringEvent>,
+    last_event_id: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct MonitoringEvent {
+    pub message: MonitoringEventPayload,
+    pub id: u32,
```

Comment:
Remove `pub` from fields.

Reply:
And access them with methods? isn't it extra code since the methods are just getters?

Reply:
It is extra code, but we will deny modifying the data from the outside, the event should be read-only. Also if all attributes are `pub`, you can create an instance of the struct from outside this module, which we also don't want to allow.